Makefile

```diff
@@ -41,7 +41,7 @@ test:
 	for test in highlighters/*; do \
 		if [ -d $$test/test-data ]; then \
 			echo "Running test $${test##*/}"; \
-			env -i QUIET=$$QUIET $${TERM:+"TERM=$$TERM"} $(ZSH) -f tests/test-highlighting.zsh "$${test##*/}"; \
+			env -i QUIET=$$QUIET $${TERM:+"TERM=$$TERM"} PATH="$$PATH" $(ZSH) -f tests/test-highlighting.zsh "$${test##*/}"; \
```

Comment:
This would leave $PATH set for the entirety of the test run, though.  How about:

```diff
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ INSTALL?=install -c
 PREFIX?=/usr/local
 SHARE_DIR?=$(DESTDIR)$(PREFIX)/share/$(NAME)
 DOC_DIR?=$(DESTDIR)$(PREFIX)/share/doc/$(NAME)
-ZSH?=zsh # zsh binary to run tests with
+ZSH?="`which zsh`" # zsh binary to run tests with
 
 all:
        cd docs && \
```

?

Reply:
Doens't work:
```
$ make test
ZSH_PATCHLEVEL=zsh-5.8-243-g5b4cfb1
Running test brackets
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
Running test main
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
Running test pattern
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
Running test regexp
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
/usr/bin/env: 'zsh': No such file or directory
make: *** [Makefile:40: test] Error 2
```
Reason: The shebang of `test-highlighting.zsh` contains `#!/usr/bin/env zsh`, i.e. another `env` call, which then again won't find `zsh` anymore if `PATH` is not set.

I guess we are better of with having `PATH` set. Potential external commands inside test-cases itself also need to be found.

Reply:
The shebang line? Really? I'd have expected it not to matter when `-c` isn't given.

How about `$(ZSH) -fc '. tests/test-highlighting.zsh` to bypass the shebang line? Would that break anything?

Reply:
> The shebang line? Really? I'd have expected it not to matter when `-c` isn't given.

On Linux on FreeBSD it doesn't matter.  What's the output of `zsh -f =(<<<$'#!/usr/bin/env date')` on Cygwin?

Reply:
Correction: It is not the shebang of `test-highlighting.zsh` but the other external scripts at are being run *from* there. E.g.
```
print > >($results_filter | $root/tests/tap-colorizer.zsh) -r -- "# global (driver) tests"
print > >($results_filter | $root/tests/tap-colorizer.zsh) -r -- "1..1"
```

What this actually means is that if ZSH is set in the Makefile, this does not guarantee that every part of the test suite is executed with that exact ZSH binary. Those called scripts might not be.

Reply:
tap-colorizer needn't be run under the same zsh version as everything else. The only reason it's written in zsh is to minimize dependencies.

Anyway, feel free to PR the fix.